.circleci/config.yml

```diff
@@ -54,10 +54,17 @@ opam-switch: &opam-switch
   steps:
     - checkout
     - run: *before_script
+    - run:
+        name: Cache selection
+        command: |
+          source ~/.profile
+          # We can't use environment variables in cache names
+          # So put it in a file and use the checksum
+          echo "$COMPILER" > COMPILER
     - restore_cache:
         keys:
-          - coq-opam-cache-v1-{{ arch }}-{{ .Environment.COMPILER }}-{{ checksum ".circleci/config.yml" }}-
-          - coq-opam-cache-v1-{{ arch }}-{{ .Environment.COMPILER }}- # this grabs old cache if checksum doesn't match
```

Comment:
This `.Environment` thing only picks up certain variables which does not include user variables like COMPILER, see https://circleci.com/docs/2.0/configuration-reference/#save_cache
So instead of getting 1 cache per compiler we get 1 overall cache. It still works correctly because we use workspace persistence to send opam to jobs after it's setup, but we have to do extra work in the opam-boot jobs because they get incorrect caches.